mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -210,31 +197,23 @@ def results(self, records):
             ([string, string, string], int): list of three colors and number of votes it received
         """
         
-        if(self.debug):
-            self.log("self.loging dictionary at start of results")
-            self.log(records)
+        self.log("fLogging dictionary at start of results. {records}")
 
         pattern_count = max(records, key=records.get)
         max_counter = records.get(pattern_count)
         
         if (max_counter):
-            if(self.debug):
-                self.log("Determined pattern based on the voting scheme from records")
-                self.log(pattern_count)
-                self.log(f"With vote of: {records.get(pattern_count)}")
-                self.log(f"{self.lookup[int(pattern_count[0])]},{self.lookup[int(pattern_count[1])]},{self.lookup[int(pattern_count[2])]}")
+            voteCount = records.get(pattern_count)
+            self.log(f"Determined pattern based on the voting scheme from records: {pattern_count}, With vote of: {voteCount}\n,
+                     {self.lookup[int(pattern_count[0])]},{self.lookup[int(pattern_count[1])]},{self.lookup[int(pattern_count[2])]},
+                     Individual colors: {color1}, {color2}, {color3}")
```

Comment:
color 1 color2 and color3 are not defined until the next lines right? Maybe place this log statement after

Reply:
Also based off the next lines arent you just printing the same thing over again? You log both the self.lookup and then the color but they are equal: 
`color1 = self.lookup[int(pattern_count[0])]`